src/validationinterface.h

```diff
@@ -35,6 +36,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {};
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {};
     virtual void BlockChecked(const CBlock&, const CValidationState&) {};
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
Yes, generate can just be changed to not need the GetScriptForMining
interface. And setgenerate could just be deleted IMHO and replaced with a
simple proof of concept Python or C miner in the tree.